pytorch_lightning/utilities/model_summary.py

```diff
@@ -430,8 +437,8 @@ def _is_lazy_weight_tensor(p: Tensor) -> bool:
 
 
 def summarize(
-    lightning_module: "pl.LightningModule", mode: Optional[str] = "top", max_depth: Optional[int] = None
-) -> Optional[ModelSummary]:
+    lightning_module: "pl.LightningModule", mode: Optional[str] = None, max_depth: Optional[int] = None
+) -> ModelSummary:
```

Comment:
this should have been the correct way in the original PR. mode is deprecated, we want to set it to None and then detect if someone sets it so we can warn appropriately.